doc/source/user_guide/reshaping.rst

```diff
@@ -719,6 +719,58 @@ To choose another dtype, use the ``dtype`` argument:
     pd.get_dummies(df, dtype=bool).dtypes
 
 
+To convert a "dummy" or "indicator" ``DataFrame``, into a categorical ``DataFrame``
+(a categorical ``Series``), for example ``k`` columns of a ``DataFrame`` containing
+1s and 0s can derive a ``DataFrame`` (a ``Series``) which has ``k`` distinct values
+:func:`~pandas.from_dummies`:
+
+.. ipython:: python
+
+   d = pd.DataFrame({"prefix_a": [0, 1, 0], "prefix_b": [1, 0, 1]})
+
+   pd.from_dummies(d)
+
+The ``k`` distinct values can also be represented be a ``dropped_first`` which
+means that no vale assigned implies a the value of the dropped value:
+
+.. ipython:: python
+
+   d = pd.DataFrame({"prefix_a": [0, 1, 0]})
+
+   pd.from_dummies(d, dropped_first="b")
+
+The function is the inverse of :func:`pandas.get_dummies <pandas.reshape.get_dummies>`.
+
+All non-dummy columns are included untouched in the output. You can control
+which columns are included in the output with the ``columns`` argument.
+
+.. ipython:: python
+
+    pd.get_dummies(df, columns=["C", "prefix_A", "prefix_B"])
+
+You can pass values for for the ``prefix_sep`` argument depending on how many or
+nested prefix separators are used in the column names. By default the prefix
+separator is assumed to be a '_', however ``prefix_sep`` can be specified in
+3 ways:
+
+* string: Use the same value for ``prefix_sep`` for each column
+  to be dencoded.
+* list: Variables will be decoded by the first instance of prefix separator passed
```

Comment:
umm this seems quite complicated. can you show the results here (in a comment)

Reply:
Sure
```python
  dummies = DataFrame(
      {
          "col1_a-a": [1, 0, 1],
          "col1_b-b": [0, 1, 0],
          "col2-a_a": [0, 1, 0],
          "col2-b_b": [1, 0, 0],
          "col2-c_c": [0, 0, 1],
      },
  )
```
```bash
>>> from_dummies(dummies, prefix_sep={"col1": "_", "col2": "-"})
             col1     col2 
        0    a-a      b_b
        1    b-b      a_a 
        2    a-a      c_c
```